### PR TITLE
feat: allow move to space and delete for sql charts

### PR DESCRIPTION
**Pull request by rephus**
Closes:  https://github.com/lightdash/lightdash/issues/11293

### Description:

Screenshot from 2024-09-03 12-13-04

- [ ] The rest of the actions are hidden 

- rename : needs new modal component
- add to dashboard: needs modifying hooks on add to dashboard component
- pin : needs new endpoint
- promote: needs new endpoint

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

**Review by joaoviana — CHANGES_REQUESTED** [review]
in `ResourceActionMenu` L~365, there's a `Menu.Divider`, can you add this?
```tsx
 <Menu.Divider
                                display={isSqlChart ? 'none' : 'block'}
                            />
```

**Review by joaoviana — CHANGES_REQUESTED** [review]
Moving to an existing space works ✅ 

Creating a space on the menu item + moving to it doesn't. 
Wonder if this is an existing bug though, checking

EDIT: Correct, this is a bug unrelated to this work, so I will ignore this in the review: https://github.com/lightdash/lightdash/issues/11363

File: packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx [hunk 1]
```diff
@@ -229,6 +243,7 @@ const ResourceViewActionMenu: FC<ResourceViewActionMenuProps> = ({
                                 item,
                             });
                         }}
+                        sx={{ display: isSqlChart ? 'none' : 'block' }}
```
Comment (joaoviana) [inline packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx:246]: could we use the `hidden` prop instead? 
```tsx
hidden={isSqlChart}
```

otherwise we have to deal with default styling if it's not a `sql chart`

image

Comment (rephus) [reply packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx:246]: hidden didn't work to me, let me try again

Comment (joaoviana) [reply packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx:246]: then we can do `disabled`?

Comment (rephus) [reply packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx:246]: https://github.com/mantinedev/mantine/issues/5521#issuecomment-1878444848

Comment (rephus) [reply packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx:246]: disabled without tooltip can be confusing, don't you think ? and I don't want to add many extra tooltips...

Comment (rephus) [reply packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx:246]: I can do this instead 
`                            sx={isSqlChart ?{ display:  'none'}: {}}`

Comment (joaoviana) [reply packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx:246]: then we can replace in the ternary the `block` with `flex` 👍

Comment (rephus) [reply packages/frontend/src/components/common/ResourceView/ResourceActionMenu.tsx:246]: no need to add any flex, we can leave it empty 

> sx={isSqlChart ?{ display:  'none'}: {}}


image

**Review by joaoviana — APPROVED** [review]

**Comment by rephus** [comment]
I'll create a ticket for the missing functions in the context menu (low priority)
